doc/compressed_transactions.md

```diff
@@ -0,0 +1,153 @@
+height and index. This requires the decompression to have access to the blockchain, and
+also means that compression is ineffective for transactions that spend unconfirmed or
+insufficiently confirmed outputs.
+
+Even without compression, Taproot keyspends are very small: as witness data they
+include only a single 64/65-byte signature and do not repeat the public key or
+any other metadata. By using pubkey recovery, we obtain Taproot-like compression
+for legacy and Segwit transactions.
+
+The main applications for this schema are for steganography, satellite/radio broadcast, and
+other low bandwidth channels with a high CPU availability on decompression. We
+assume users have some ability to shape their transactions to improve their
+compressibility, and therefore give special treatment to certain transaction forms.
+
+## 2. Methods
+
+The four main methods to achieve a lower transactions size are:
+
+1. packing transaction metadata before the transaction and each of its inputs and
+outputs to determine the structure of the following data.
+2. replacing 32-bit numeric values with variable-length integers (VarInts).
+3. using compressed signatures and public key recovery upon decompression.
+4. replacing the 36-byte txid/vout pair with a blockheight and output index.
+
+Method 4 will cause the compressed transaction to be undecompressable if a block
+reorg occurs at or before the block it's included in. Therefore, we'll only compress
+the Txid if the transaction input is at least one hundred blocks old.
+
+## 3 Schema
+
+### 3.1 Primitives
+
+| Name             | Width     | Description |
+|------------------|-----------|-------------|
+| VarInt           | 1-5 Bytes | For 0-253, encode the value directly in one byte. For 254-65535, encode 254 followed by 2 little-endian bytes. For 65536-(2^32-1), encode 255 followed by 4 little-endian bytes. This is the same varint used to length-prefix strings in the ordinary transaction encoding. |
+| VarInt Flag      | 2 Bits    | 1, 2 or 3 indicate literal values. 0 indicates that the value will be encoded in a later VarInt. |
+| Bitstream VarInt | Variable  | 7-bit little-endian encoding, with each 7-bit word encoded in a byte. The highest bit of each byte is 1 if more bytes follow, and 0 for the last byte. |
+| VLP-Bytestream   | Variable  | A VarInt Length Prefixed Bytestream, Has a VarInt prefixed to determine the length. |
+
+### 3.2 General Schema
+
+| Name                  | Width           | Description |
+| ----------------------|----------------|-------------|
+| Transaction Metadata  | 1-2 Bytes      | Information on the structure of the transaction. See section 3.2. |
+| Minimum Blockheight | Bitstream VarInt | Transaction locktime and input blockheights are given as offsets from this value. |
+| Version               | Variable       | If the Version was greater then 3 then this is an optional VarInt containing the Version. |
```

Comment:
In ddde5c7defae80058c35bcd5fdc27ee57935495f:

Most of these `Variables` can be more specific. e.g. varints should be `1-5 bytes` or just `VarInt`. We should use `Variable` only for strings that might conceivably be any length.

Reply:
Fixed, Thanks!